### PR TITLE
Deploy review app with 'deploy' label

### PR DESCRIPTION
**Pull request by AbigailMcP**
### Context

We want to be able to easily product review PRs by adding a 'deploy' label.
The deploy label should deploy a review app in the review env.

### Changes proposed in this pull request

- Updates to terraform config:
  - create` application.tf` for web application and application configuration
  - create `data.tf` for KV secrets (app and infra)
  - create `databases.tf` for postgres config
  - update `variables.tf`
- Updates to makefile:
  - Add auto-approve option so that `terraform-apply` and `terraform-destroy` can run in CI
  - Add `install-konduit` command for access to the backend
  - Add commands to print/edit app/infra secrets
  - Add `logs` command to tail app logs
- New `deploy-review-app` job added to the `build-and-deploy` workflow
  - This uses a new `deploy` action which deploys app and installs konduit
- New `delete-review-app` workflow which deletes review app on PR close or merge

### Guidance to review

- Add deploy label
- Check deployed app
- Close PR and check that the deployed app is destroyed

### Link to Trello card

https://trello.com/c/cSNsmW8m/63-base-aks-config-review-apps

### Checklist

- [x] Attach to Trello card
- [x] Rebased main
- [x] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

**Comment by AbigailMcP** [comment]
hmm closing the PR doesn't seem to delete the app - https://github.com/DFE-Digital/check-childrens-barred-list/actions/runs/5357813169/jobs/9719221253

File: .github/workflows/delete-review-app.yml [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+          # tf_vars_file=terraform/aks/config/review.tfvars.json
+          # echo "key_vault_name=$(jq -r '.key_vault_name' ${tf_vars_file})" >> $GITHUB_ENV
+          # echo "key_vault_app_secret_name=$(jq -r '.key_vault_app_secret_name' ${tf_vars_file})" >> $GITHUB_ENV
+          # echo "key_vault_infra_secret_name=$(jq -r '.key_vault_infra_secret_name' ${tf_vars_file})" >> $GITHUB_ENV
+      - name: Use Terraform ${{ env.TERRAFORM_VERSION }}
+        uses: hashicorp/setup-terraform@v2
+        with:
+          terraform_version: ${{ env.TERRAFORM_VERSION }}
+
+      # - name: Validate Azure Key Vault secrets
```
Comment (gpeng) [inline .github/workflows/delete-review-app.yml:48]: 🤔

File: terraform/aks/cluster_data.tf [hunk 2]
```diff
@@ -0,0 +1,4 @@
+module "cluster_data" {
+  source = "./vendor/modules/aks//aks/cluster_data"
+  name   = var.cluster
+}
```
Comment (gpeng) [inline terraform/aks/cluster_data.tf:4]: Missing newline

File: .github/workflows/delete-review-app.yml [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+      - name: Update ${{ matrix.environment }} status
+        id: deactivate-env
+        if: ${{ always() && env.TF_STATE_EXISTS == 'true' }}
+        uses: bobheadxi/deployments@v1
+        with:
+          step: deactivate-env
+          token: ${{ secrets.GITHUB_TOKEN }}
+          env: review-${{ env.PR_NUMBER }}
+
+      # - name: Delete review-${{ env.PR_NUMBER }} environment
```
Comment (gpeng) [inline .github/workflows/delete-review-app.yml:82]: 🤔

File: .github/actions/deploy/action.yml [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+        APP_NAME: ${{ env.APP_NAME }}
+        TF_VAR_azure_sp_credentials: ${{ inputs.azure-credentials }}
+        CONFIRM_PRODUCTION: true
+
+    - name: Set up environment variables
+      shell: bash
+      run: |
+        case ${{ env.cluster }} in
+          test)
+            echo "cluster_rg=s189t01-tsc-ts-rg" >> $GITHUB_ENV
```
Comment (gpeng) [inline .github/actions/deploy/action.yml:92]: I wonder could we extract these and the URLs up there ^ (42,52 etc). I think this file would just be generic then.

**Review by RMcVelia — APPROVED** [review]

**Review by gpeng — APPROVED** [review]

**Pull request merged**